MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -538,6 +577,373 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputHermiteSplineModel(vtkMRMLMarkups
 }
 
 //------------------------------------------------------------------------------
+void vtkSlicerMarkupsToModelLogic::ComputePointParametersRawIndices(vtkPolyData * markupsPointsPolyData, vtkDoubleArray* markupsPointsParameters)
+{
+  vtkPoints* points = markupsPointsPolyData->GetPoints();
+  if (points == NULL)
+  {
+    vtkErrorMacro("Markups contains a vtkPoints object that is null.");
+    return;
+  }
+
+  int numPoints = points->GetNumberOfPoints();
+  // redundant error checking, to be safe
+  if (numPoints < LINE_MIN_NUMBER_POINTS)
+  {
+    vtkErrorMacro("Not enough points to compute polynomial parameters. Need at least " << LINE_MIN_NUMBER_POINTS << " points but " << numPoints << " are provided.");
+    return;
+  }
+
+  if (markupsPointsParameters->GetNumberOfTuples())
+  {
+    // this should never happen, but in case it does, output a warning
+    vtkWarningMacro("markupsPointsParameters already has contents. Clearing.");
+    while (markupsPointsParameters->GetNumberOfTuples()) // clear contents just in case
+      markupsPointsParameters->RemoveLastTuple();
+  }
+
+  for (int v = 0; v < numPoints; v++)
+  {
+    markupsPointsParameters->InsertNextTuple1( v / double(numPoints-1) );
+    // division to clamp all values to range 0.0 - 1.0
+  }
+}
+
+//------------------------------------------------------------------------------
```

Comment:
This class is getting really big. Please move polynomial fitting code to another class. Maybe create a http://www.vtk.org/doc/nightly/html/classvtkParametricFunction.html and move the algorithms into that. You may even contribute it to VTK or as a VTK external module in the future.